lib/compiler/passes/generate-js.js

```diff
@@ -721,17 +757,17 @@ function generateJS(ast, options) {
             break;
           }
 
-          case op.ACCEPT_N:           // ACCEPT_N n
-            parts.push(stack.push(
-              bc[ip + 1] > 1
-                ? "input.substr(peg$currPos, " + bc[ip + 1] + ")"
-                : "input.charAt(peg$currPos)"
-            ));
-            parts.push(
-              bc[ip + 1] > 1
-                ? "peg$currPos += " + bc[ip + 1] + ";"
-                : "peg$currPos++;"
+          case op.MATCH_UNICODE_CLASS: { // MATCH_UNICODE_CLASS c, a, f, ...
+            const regNum = bc[ip + 1];
+            compileInputChunkCondition(
+              inputChunk => `${r(regNum)}.test(${inputChunk})`, 1, -1
             );
+            break;
+          }
+
+          case op.ACCEPT_N:           // ACCEPT_N n
+            parts.push(stack.push(getChunkCode(bc[ip + 1])));
+            parts.push(getIncrCode(bc[ip + 1], stack.top()));
```

Comment:
I'm still only 90% sure that `stack.top` is right here, even after testing it a good amount.  I don't understand the bytecodes as much as I'd like.